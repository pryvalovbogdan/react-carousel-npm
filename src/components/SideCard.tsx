import React from 'react';

import stylesCss from '../styles/carousel.module.css';
import { renderChildren } from '../utils';
import { ISideCard } from '../types/CarouselTypes';

const SideCard = ({ child, cardWidth, item, width, index, selected, widthCard, marginCard = 1 }: ISideCard) => {
  const marginRight = width < 900 ? marginCard / 2 : marginCard * (selected - 1);
  const marginLeft = width < 900 ? marginCard / 2 : marginCard;

  return (
    <div className={stylesCss['side-card-box']} style={{ right: index === 0 ? 'auto' : 0 }} key={item.key}>
      {renderChildren(child, {
        ...item,
        ...styles.card,
        maxWidth: cardWidth,
        style: styles.sideCardParent,
      })}
      {renderChildren(child, {
        ...item,
        style: {
          ...styles.sideCardChild,
          ...styles.card,
          maxWidth: cardWidth,
          right:
            index === 0
              ? `calc(100% + ${marginLeft}px`
              : width - (selected - 1) * widthCard - widthCard - marginRight + 'px',
        },
      })}
    </div>
  );
};

export default SideCard;

const styles = {
  card: {
    width: '100%',
  },
  sideCardParent: {
    width: '30%',
    height: '0',
    overflow: 'hidden',
    border: 'none',
  },
  sideCardChild: {
    position: 'absolute',
    top: 0,
  },
};
