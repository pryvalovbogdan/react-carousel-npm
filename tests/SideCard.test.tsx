import { render } from '@testing-library/react';

import React from 'react';
import SideCard from '../src/components/SideCard';

describe('Test SideCard', () => {
  const child = <div>Child content</div>;
  const cardWidth = 200;
  const item = { key: 'item-key', title: 'Item title' };
  const width = 800;
  const index = 1;
  const selected = 3;
  const widthCard = 150;
  const marginCard = 10;

  test('snapshot Notes', () => {
    const { baseElement } = render(
      <SideCard
        child={child}
        cardWidth={cardWidth}
        item={item}
        width={width}
        index={index}
        selected={selected}
        widthCard={widthCard}
        marginCard={marginCard}
      />,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
