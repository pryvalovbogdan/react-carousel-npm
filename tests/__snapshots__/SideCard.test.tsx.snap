// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SideCard snapshot Notes 1`] = `
<body>
  <div>
    <div
      class="side-card-box"
      style="right: 0px;"
    >
      <div
        maxwidth="200"
        style="width: 30%; height: 0px; overflow: hidden;"
        title="Item title"
        width="100%"
      >
        Child content
      </div>
      <div
        style="position: absolute; top: 0px; width: 100%; max-width: 200px; right: 345px;"
        title="Item title"
      >
        Child content
      </div>
    </div>
  </div>
</body>
`;
